drivers/opencl/src/main/java/uk/ac/manchester/tornado/drivers/opencl/graal/phases/TornadoFPGAPragmas.java

```diff
@@ -44,17 +38,30 @@
 import org.graalvm.compiler.nodes.loop.LoopsData;
 import org.graalvm.compiler.options.OptionValues;
 import org.graalvm.compiler.phases.BasePhase;
-
+import uk.ac.manchester.tornado.api.TornadoDeviceContext;
 import uk.ac.manchester.tornado.api.enums.TornadoDeviceType;
-import uk.ac.manchester.tornado.drivers.opencl.graal.nodes.PragmaUnrollNode;
+import uk.ac.manchester.tornado.drivers.opencl.graal.nodes.IntelUnrollPragmaNode;
+import uk.ac.manchester.tornado.drivers.opencl.graal.nodes.XilinxPipeliningPragmaNode;
 import uk.ac.manchester.tornado.runtime.graal.nodes.TornadoLoopsData;
 import uk.ac.manchester.tornado.runtime.graal.phases.TornadoHighTierContext;
 
-public class TornadoPragmaUnroll extends BasePhase<TornadoHighTierContext> {
+import java.util.List;
+
+import static org.graalvm.compiler.core.common.GraalOptions.MaximumDesiredSize;
+import static org.graalvm.compiler.nodes.loop.DefaultLoopPolicies.Options.ExactFullUnrollMaxNodes;
+import static org.graalvm.compiler.nodes.loop.DefaultLoopPolicies.Options.FullUnrollMaxNodes;
+
+public class TornadoFPGAPragmas extends BasePhase<TornadoHighTierContext> {
 
     private static final int UNROLL_FACTOR_NUMBER = 4;
+    private static final int XILINX_PIPELINING_II_NUMBER = 1;
```

Comment:
Can you add documentation of this flag? I think a link to the Xilinx docs will also help.

Reply:
Why is the value `1`?

Reply:
Yes, sure. I decided to add this constant so that to have the opportunity in future to configure it. 

Currently, the Xilinx HLS frameworks (SDAccel, Vitis) they have as default value to be 1, which aims to pipeline all iterations of the loop, if possible.

Here is a link: https://www.xilinx.com/html_docs/xilinx2020_2/vitis_doc/openclattributes.html#sgo1504034359903

I will add a comment for these two constants.